File: cpp/3. Basic math/6_prime.cpp
#include<bits/stdc++.h>
using namespace std;
bool prime(int n)
{
    int cnt=0;
    int a=sqrt(n);
    for(int i=1;i<=a;i++) //int i=1;i<=sqrt(n);i++
    {
        if(n%i==0)
        {
            cnt=cnt+1;
            if((n/i)!=i)
            {
             cnt=cnt+1;
            }
        }
        
    }
    if(cnt==2)
    {
        return true;
    }
    else
    {
        return false;
    }
  
    

}
int main()
{
    int n;
    cout<<"enter number:";
    cin>>n;
    bool ans=prime(n);
    if(ans)
    {
        cout<<n<<" is prime number";
    }
    else
    {
         cout<<n<<" is not prime number";
    }
    return 0;
}

File: python/2. Basic Math/6_prime.py
import math
def checkPrime(n):
    cnt=0
    sqrtN=int(math.sqrt(n))
    for i in range(1,sqrtN+1):
        if n%i==0:
            cnt+=1
            if i!=n//i:
                cnt+=1
    if cnt==2:
        return True
    else:
        return False
    

    n = 1483
    isPrime = checkPrime(n)
    if isPrime:
        print(n, "is a prime number.")
    else:
        print(n, "is not a prime number.")

